src/test/lib/utils/list_directory_test.cpp

```diff
@@ -0,0 +1,45 @@
+#include "utils/list_directory.hpp"
+
+#include <sys/stat.h>
+#include <filesystem>
+#include <fstream>
+
+#include "base_test.hpp"
+
+namespace hyrise {
+
+class ListDirectoryTest : public BaseTest {};
+
+TEST_F(ListDirectoryTest, ListBasicDirectory) {
+  // Setup: Creates test-directory in the folder, the test is executed in.
+  const auto TEST_DIRECTORY = std::string("./test-directory");
+
+  struct stat s;
+  auto directory_exists = (stat(TEST_DIRECTORY.c_str(), &s) == 0);
+  if (!directory_exists) {
+    directory_exists = std::filesystem::create_directory(TEST_DIRECTORY);
+  }
+
+  EXPECT_TRUE(directory_exists);
+
+  const auto files = std::vector<std::string>{TEST_DIRECTORY + "/one.test", TEST_DIRECTORY + "/two.test",
+                                              TEST_DIRECTORY + "/three.test", TEST_DIRECTORY + "/four.test"};
+
+  for (const auto& file : files) {
+    std::ofstream{file};
+  }
+  std::filesystem::create_directory(TEST_DIRECTORY + "/test-subdirectory");
```

Comment:
1. This should be in `SetUp()`, with a `_directory_path` member. Even though you only have a single test case, it separates more clearly what you actually test.
2. Please also look at `base_test_hpp`, where we have a `test_data_path` (in which your directory should be created.
3. Please have a look at https://en.cppreference.com/w/cpp/filesystem/create_directory about what `create_directory` does and remove everything you do with `stat` here.